pkg/storage/tpr/storage_interface.go

```diff
@@ -86,18 +87,37 @@ func (t *store) Create(
 	ttl uint64,
 ) error {
 
-	ns, _, err := t.decodeKey(key)
+	ns, name, err := t.decodeKey(key)
 	if err != nil {
 		glog.Errorf("decoding key %s (%s)", key, err)
 		return err
 	}
 
+	// Check if this resource already exists
+	req := t.cl.Get().AbsPath(
+		"apis",
+		groupName,
+		tprVersion,
+		"namespaces",
+		ns,
+		t.singularKind.URLName(),
+		name,
+	)
+	res := req.Do()
+	if res.Error() != nil {
+		glog.Errorf("executing GET for %s/%s (%s)", ns, name, err)
+	}
+	var statusCode int
+	res.StatusCode(&statusCode)
+	if statusCode != http.StatusNotFound {
+		return storage.NewKeyExistsError(key, 0)
+	}
```

Comment:
I think we've discussed this before. I'm not sure what we can possible do about that. The etcd3-based storage impl accounts for that easily because etcd supports transactions. The core apiserver does not. Thoughts?